modules/locking/src/main/frontend/src/locking/SubjectLockAction.jsx

```diff
@@ -0,0 +1,366 @@
+  Typography
+} from "@mui/material";
+import Alert from '@mui/material/Alert';
+import LockIcon from "@mui/icons-material/Lock";
+import LockOpenIcon from "@mui/icons-material/LockOpen";
+import AssignmentIcon from "@mui/icons-material/Assignment";
+import { DateTime } from "luxon";
+
+import { fetchWithReLogin, GlobalLoginContext } from "../login/loginDialogue.js";
+import ErrorDialog from "../components/ErrorDialog.jsx";
+
+function SubjectLockAction(props) {
+  const {
+    subject,
+    reloadSubject,
+    size,
+    variant = "icon",
+    className
+  } = props;
+
+  const METHOD_LOCK = "LOCK";
+  const METHOD_UNLOCK = "UNLOCK";
+  const ACTION_CONTINUE = "CONTINUE";
+  const ACTION_LOCK = "LOCK";
+  const ACTION_UNLOCK = "UNLOCK";
+
+  const [ open, setOpen ] = useState(false);
+  const [ displayButton, setDisplayButton ] = useState(true);
+  const [ dialogTitle, setDialogTitle ] = useState(null);
+  const [ dialogHeader, setDialogHeader ] = useState(null);
+  const [ dialogContent, setDialogContent ] = useState(null);
+  const [ nextAction, setNextAction ] = useState(ACTION_CONTINUE);
+  const [ errorMessage, setErrorMessage ] = useState("");
+  const [ requestInProgress, setRequestInProgress ] = useState(false);
+  const [ isLocked, setLocked ] = useState(false);
+  const [ actionContent, setActionContent ] = useState(null);
+  const [ actionLabel, setActionLabel ] = useState("");
+
+  const globalLoginDisplay = useContext(GlobalLoginContext);
+
+  let lockUnlockText = isLocked ? "Unlock" : "Lock";
+  let entryType = subject?.type?.label || "Subject";
+  let entryPath = subject?.["@path"];
+
+  useEffect(() => {
+    setLocked(subject?.statusFlags && subject.statusFlags.includes("LOCKED"));
+  }, [subject['jcr:lastModified']])
+
+  useEffect(() => {
+    // Hide this button if this subject has a parent and the parent is locked since
+    //  current subject can't be locked or unlocked
+    setDisplayButton(!subject?.parents?.["cards:lock"]);
+  }, [subject?.parents?.['jcr:lastModified']])
+
+  let openDialog = () => {
+    setDialogContent(null);
+    setActionContent(null);
+    if (isLocked) {
+      handleOpenDialogLocked();
+    } else {
+      handleOpenDialogUnlocked();
+    }
+    // TODO: move to a global context?
+    // See: AdminNavbarLinks
+    setOpen(true);
+  }
+
+  let closeDialog = () => {
+    setOpen(false);
+  }
+
+  let openError = (message) => {
+    closeDialog();
+    setErrorMessage(message);
+  }
+
+  let closeError = () => {
+    setErrorMessage("");
+  }
+
+  let handleOpenDialogUnlocked = () => {
+    setDialogTitle(`Sign off and Lock ${entryType} "${subject?.identifier}"`);
+    setNextAction(ACTION_CONTINUE);
+    setDialogContent(null);
+    fetchIncompleteForms();
+  }
+
+  let getLockWarning = () => {
+    return <Alert severity="warning">Once you sign off on this {entryType}, it will be locked along with
+            all the associated data forms and no further edits will be possible. Proceed?</Alert>
```

Comment:
Slightly easier to read:
```suggestion
    return (
      <Alert severity="warning">
        Once you sign off on this {entryType}, it will be locked along with
        all the associated data forms and no further edits will be possible. Proceed?
      </Alert>
    );
```